hotel-management/src/components/RoomForm/index.tsx

```diff
@@ -69,10 +75,29 @@ export const RoomForm = ({ room: initialRoom }: { room: Room }) => {
     }
   };
 
-  const handleSubmit = (event: FormEvent<HTMLFormElement>) => {
+  const addRoom = async () => {
+    const { ROOMS } = API_ENDPOINT;
+    const { POST_ROOM_SUCCESS, POST_ROOM_FAILED } = REQUEST_MESSAGE;
+    const newRoom = { ...room };
+    delete newRoom.id;
+
+    setIsLoading(true);
+    const { error } = await HttpService.post<Room>('ROOMS', newRoom);
+
+    if (error) {
+      toast?.add(POST_ROOM_FAILED, 'error');
+    } else {
+      toast?.add(POST_ROOM_SUCCESS, 'success');
```

Comment:
Why we use optional chaining here?

Reply:
Because the default value for the context of toast is null, and I use it for a case when a component using the `useToast` hook isn't wrapped in `ToastContextProvider`